backend/src/routes/api/nim-serving/index.ts

```
@@ -1,40 +1,56 @@
+import { secureAdminRoute } from '../../../utils/route-security';
 
 export default async (fastify: KubeFastifyInstance): Promise<void> => {
+  const resourceMap: Record<string, { type: 'Secret' | 'ConfigMap'; path: string[] }> = {
+    apiKeySecret: { type: 'Secret', path: ['spec', 'apiKeySecret', 'name'] },
+    nimPullSecret: { type: 'Secret', path: ['status', 'nimPullSecret', 'name'] },
+    nimConfig: { type: 'ConfigMap', path: ['status', 'nimConfig', 'name'] },
+  };
+
   fastify.get(
     '/:nimResource',
-    async (
-      request: OauthFastifyRequest<{
-        Params: { nimResource: string };
-      }>,
-    ) => {
-      logRequestDetails(fastify, request);
-      const { nimResource } = request.params;
-      const { coreV1Api, namespace } = fastify.kube;
+    secureAdminRoute(fastify)(
+      async (request: OauthFastifyRequest<{ Params: { nimResource: string } }>) => {
+        logRequestDetails(fastify, request);
+        const { nimResource } = request.params;
+        const { coreV1Api, namespace } = fastify.kube;
 
-      if (secretNames.includes(nimResource)) {
-        try {
-          return await coreV1Api.readNamespacedSecret(nimResource, namespace);
-        } catch (e) {
-          fastify.log.error(`Failed to fetch secret ${nimResource}: ${e.message}`);
-          throw createCustomError('Not found', 'Secret not found', 404);
+        // Fetch the Account CR to determine the actual resource name dynamically
+        const account = await getNIMAccount(fastify, namespace);
+        if (!account) {
+          throw createCustomError('Not found', 'NIM account not found', 404);
+        }
+
+        const resourceInfo = resourceMap[nimResource];
+        if (!resourceInfo) {
+          throw createCustomError('Not found', `Invalid resource type: ${nimResource}`, 404);
+        }
+
+        // Get the resource name
+        const resourceName = resourceInfo.path.reduce((acc, key) => acc?.[key], account as any);
```

`as any` should be removed / fixed because this could be hiding an actual error